native/core/src/execution/datafusion/planner.rs

```diff
@@ -623,6 +624,21 @@ impl PhysicalPlanner {
                 let child = self.create_expr(expr.child.as_ref().unwrap(), input_schema.clone())?;
                 Ok(Arc::new(GetStructField::new(child, expr.ordinal as usize)))
             }
+            ExprStruct::CreateArray(expr) => {
+                let values = expr
+                    .values
+                    .iter()
+                    .map(|expr| self.create_expr(expr, input_schema.clone()))
+                    .collect::<Result<Vec<_>, _>>()?;
+                let data_type = to_arrow_datatype(expr.datatype.as_ref().unwrap());
+
+                Ok(Arc::new(ScalarFunctionExpr::new(
+                    "make_array",
+                    make_array_udf(),
+                    values,
+                    data_type,
+                )))
```

Comment:
I'm not sure if this is the right/best way to make use of a ScalarUDF in DataFusion, but it compiled and seemed to work

Reply:
For calling scalar udfs, you can take a look at `scalarExprToProtoWithReturnType` or `scalarExprToProto` in `QueryPlanSerde`

By using these, we can omit `planner` change or `proto` definition.

Reply:
Thanks, updated to use `scalarExprToProtoWithReturnType`. Had to register the nested functions in the functions registry as well.